tests/integration/viewer.js

```diff
@@ -25,9 +26,104 @@ function setDateTimeValue (selector, value) {
 	});
 }
 
+function resetDateTimeValue (selector) {
+	return browser.elements(selector).then(function(res) {
+		const self = browser;
+		const elementIdValueCommands = res.value.map(function(elem) {
+			return self.elementIdValue(elem.ELEMENT, '');
+		});
+
+		return this.unify(elementIdValueCommands, {
+			extractValue: true
+		});
+	});
+}
+
+// go to the admin page set a url
+function addItem(url, duration, scheduledTime) {
+
+	// 0 or undefined are not valid durations
+	duration = duration || 60;
+
+	return tabs.admin()
+		.waitForExist('#chkscreen-12345')
+		.waitForExist('label[for=chkscreen-12345]')
+		.isSelected('#chkscreen-12345')
+		.then(tick => {
+			if (!tick) return browser.click('label[for=chkscreen-12345]');
+		})
+		.click(`#selurlduration option[value="${duration}"]`)
+		.then(function () {
+			if (scheduledTime) return setDateTimeValue('#time', scheduledTime);
+			return resetDateTimeValue('#time');
+		})
+		.setValue('#txturl', url)
+		.then(() => console.log(`
+Setting Url: ${url}
+Duration: ${duration}
+Scheduled: ${scheduledTime}`))
+		.click('#btnsetcontent');
+}
+
+// go to the admin page pop off the top of the queue
+function removeItem(url) {
+	const xSelector = `.queue li[data-url="${url}"] .action-remove`;
+
+	return tabs.admin()
+		.waitForExist(xSelector)
+		.click(xSelector);
+}
+
+function printLogOnError(e) {
+
+	// show browser console.logs
+	return logs()
+	.then(function () {
+		console.log(e.message);
+		process.exit(1);
```

Comment:
I think that may leave the selenium and application servers running as they are detached child processes. These get killed in the wdio file.